README.md

```diff
@@ -77,11 +106,130 @@ spec:
     gcpProjectID: example-project-id
 ```
 
-## Running the tests
+### Importing your own secrets
+
+In addition to generating secrets, the `secret-agent` allow users to import their own secrets. This is especially useful for things like certificates and certificate authorities that can be referenced by other secrets in the SAC. For example, a user can import their organization's CA and use it to sign certificates generated by the `secret-agent`.
+
+All that is required is to provide a Kubernetes secret with the same name and same key names as described in the SAC. It is important to note that if the [cloud backup](#enabling-cloud-backup) feature is enabled, the secret to be imported must be provided using the cloud manager's secret manager. The `secret-agent` will ignore local secrets if [cloud backup](#enabling-cloud-backup) is enabled.
+
+### Naming Convention For Cloud Backups
+
+There is a naming convention used by `secret-agent` to store and read secrets from the cloud secret managers. In general, the names follow the format:
+
+```bash
+$prefix-$namespace-$secretName-$keyName [If secretsManagerPrefix is provided]
+$namespace-$secretName-$keyName [If no prefix is provided]
+```
+
+Due to cloud provider limitations, all `/`, `.` and `_` characters in secret names are replaced by `-` when accessing the cloud secret managers.
+
+For example, consider the following secret agent configuration:
+
+```yaml
+---
+apiVersion: secret-agent.secrets.forgerock.io/v1alpha1
+kind: SecretAgentConfiguration
+metadata:
+  name: forgerock-sac
+  namespace: dev
+spec:
+  appConfig:
+    secretsManagerPrefix: "devCluster"
+    awsRegion: us-east-1
+    secretsManager: AWS
+  secrets:
+  - name: ds-passwords
+    keys:
+      - name: dirmanager.pw
+        type: password
+```
 
-* Some of the tests exercise parts of the code that os.Exec openssl and keytool, and kubebuilder's etcd
-  * It's easiest to test in Docker
-  * Ensure you're kube context is hooked up to a test cluster, such as minikube, then
-    * `docker build -t gcr.io/forgerock-io/secret-agent-testing:latest -f --target=tester .`
-    * `docker run -it --rm -v ${PWD}:/root/go/src/github.com/ForgeRock/secret-agent gcr.io/forgerock-io/secret-agent-testing:latest`
-    * `go test ./...`
+The secret generated by the this SAC would be stored as `devCluster-dev-ds-passwords-dirmanager-pw` in the AWS Secret Manager. The same name would apply to other cloud providers.
+
+Some key types require more than one secret to be backed up. Such key types require a public and private components stored separately. This is the case for key types: `ca`, `keypair`, `ssh` and `keytool`. These secrets use the same naming convention described above and append a suffix to the main name as constructed previously.
+
+Key Type | Name
+--- | ---
+`ca`      | $NAME-pem<br>$NAME-private-pem
+`keypair` | $NAME-pem<br>$NAME-private-pem
+`ssh`     | $NAME<br>$NAME-pub
+`keytool` | $NAME<br>$NAME-storepass<br>$NAME-keypass
```

Comment:
do we want to note the max size in AWS of 65k? or maybe below in some of the config?